lib/internal/test_runner/harness.js

```diff
@@ -217,6 +219,18 @@ async function startSubtest(subtest) {
 }
 
 function runInParentContext(Factory) {
+  function validateTestOptions(options) {
+    const validOptions = ['concurrency', 'only', 'signal', 'skip', 'todo', 'timeout'];
```

Comment:
this should also be defined at module level so you don’t need a new array every time - and if it’s a SafeSet instead of an array it’ll be faster and simpler to do the lookup - you won’t need primordials.

Reply:
Did I do it correctly?

Reply:
shouldn't this actually validate against a schema instead of just looking for accepted options?

for example, if I do `concurrency: 'foo'` or `concurrency: Infinity` it should throw

Reply:
Good call; yes, it should check the types of the input too.

Reply:
That check is already covered as far as I saw. The only issue I saw was that it didn't check for what keys were passed to the options.

But if I am wrong, do let me know so that I can look into it.

Reply:
> Good call; yes, it should check the types of the input too.

Alright then, I'll work on it

Reply:
In addition to validating that only accepted options are provided (which you have so far) - it should also perform strict type checking as per defined in the [docs](https://nodejs.org/dist/latest-v21.x/docs/api/test.html#testname-options-fn):

```
concurrency: <number> | <boolean>
only: <boolean>
signal: <AbortSignal>
skip: <boolean>
todo: <boolean> | <number>
```

In addition to this, I would argue that since `Infinity` is technically a valid number value... it should probably throw an error when you try `concurrency: Infinity`

Reply:
Not my call for that question, but I'd suggest the latter, since that matches "max available".

Reply:
So actually handling of a `Infinity` value is already treated as a not-accepted value for this via https://github.com/nodejs/node/blob/main/lib/internal/validators.js#L142

Reply:
I don't think I covered the possibility of infinity. But I did a basic type check for all the keys. Is that going to be okay? Do I need to cover it too?